### PR TITLE
create todo

### PR DESCRIPTION
**Pull request by SnezhkoAndrey**

### CODE REVIEW

File: src/components/ToDoForm/ToDoForm.module.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    height: 30px;
+}
+
+.todoForm button {
+    width: 10%;
+    height: 34px;
+}
+
+.todoForm button:hover {
+    cursor: pointer;
```
Comment (aspasionov) [inline src/components/ToDoForm/ToDoForm.module.css]: move "cursor: pointer" to "todoForm button"

File: src/components/ToDoList/ToDoList.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        className={props.todo.complete ? `${style.task} ${style.complete}` : style.task}
+        onClick={() => {
+          props.completeTask(props.todo.id)
+        }}
+      >
+        {props.todo.task}
+      </div>
+      <div
+        className={
+          props.todo.complete ? `${style.deleteButton} ${style.complete}` : style.deleteButton
```
Comment (aspasionov) [inline src/components/ToDoList/ToDoList.tsx]: move this condition to variable

File: src/components/Todo/ToDo.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      }
+      setTodos([...todos, newTodo])
+    }
+  }
+
+  const removeTask = (id: number): void => {
+    setTodos([...todos.filter((td) => td.id !== id)])
+  }
+  const completeTask = (id: number): void => {
+    setTodos([...todos.map((td) => (td.id === id ? { ...td, complete: !td.complete } : { ...td }))])
```
Comment (aspasionov) [inline src/components/Todo/ToDo.tsx]: try setTodos([...todos.map((td) => (td.id === id ? { ...td, complete: !td.complete } : td))])

File: src/components/Todo/ToDo.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      <div>
+        <ToDoForm addTask={addTask} />
+      </div>
+      <div>
+        {todos.map((td) => {
+          return (
+            <ToDoList todo={td} key={td.id} removeTask={removeTask} completeTask={completeTask} />
+          )
+        })}
+      </div>
```
Comment (aspasionov) [inline src/components/Todo/ToDo.tsx]: you can reduce this function {todos.map((td) => <ToDoList todo={td} key={td.id} removeTask={removeTask} completeTask={completeTask} />}

File: src/App.tsx [hunk 5]
```diff
@@ -1,26 +1,16 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React from 'react'
+import './App.css'
+import ToDo from './components/Todo'
+import { Context } from './context/context'
```
Comment (SerhiienkoRuslan) [inline src/App.tsx]: try more understandable names

contex
  GlobaleContext.ts

File: src/context/context.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import React from 'react'
+
+export const Context = React.createContext([{ id: 1, task: '', complete: false }])
```
Comment (SerhiienkoRuslan) [inline src/context/context.ts]: export const GlobalContext = createContext<ContextType>()

export const GlobalContextProvider: FC = ({ children }) => {
const [value, setValue] = useState('')

const providerValue = {
    value,
    setValue
  }
  return <Provider value={providerValue}>{children}</Provider>
}

File: src/App.tsx [hunk 7]
```diff
@@ -1,26 +1,16 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    <Context.Provider value={[]}>
```
Comment (SerhiienkoRuslan) [inline src/App.tsx]: <GlobalContextProvider>
  <div className='App'>
        <ToDo />
      </div>
</GlobalContextProvider>

Comment (SerhiienkoRuslan) [reply src/App.tsx]: < GlobalContextProvider ></GlobalContextProvider>

File: src/components/ToDoForm/ToDoForm.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import style from './ToDoForm.module.css'
+
+interface PropsType {
+  addTask: (newValue: string) => void
+}
+
+const ToDoForm: React.FC<PropsType> = (props) => {
```
Comment (SerhiienkoRuslan) [inline src/components/ToDoForm/ToDoForm.tsx]: const ToDoForm: React.FC<PropsType> = ({ addTask }) => {

File: src/components/ToDoForm/ToDoForm.tsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  const [inputValue, setInputValue] = useState('')
+
+  const handleChange = (e: React.FormEvent<HTMLInputElement>): void => {
+    return setInputValue(e.currentTarget.value)
+  }
+
+  const handleSubmit = (e: React.SyntheticEvent): void => {
+    e.preventDefault()
+    props.addTask(inputValue)
```
Comment (SerhiienkoRuslan) [inline src/components/ToDoForm/ToDoForm.tsx]: addTask(inputValue)

File: src/components/ToDoList/ToDoList.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  removeTask: (newId: number) => void
+  completeTask: (newId: number) => void
+}
+
+const ToDoList: React.FC<PropsType> = (props) => {
+  return (
+    <div key={props.todo.id} className={style.todoList}>
+      <div
+        title={props.todo.complete ? 'click if not complete' : 'click if complete'}
+        className={props.todo.complete ? `${style.task} ${style.complete}` : style.task}
```
Comment (SerhiienkoRuslan) [inline src/components/ToDoList/ToDoList.tsx]: try this

        className={`${style.task} ${props.todo.complete ? style.complete : ''}`}

File: src/components/ToDoList/ToDoList.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  todo: {
+    id: number
+    task: string
+    complete: boolean
+  }
+  removeTask: (newId: number) => void
+  completeTask: (newId: number) => void
+}
+
+const ToDoList: React.FC<PropsType> = (props) => {
```
Comment (SerhiienkoRuslan) [inline src/components/ToDoList/ToDoList.tsx]: const ToDoList: React.FC<PropsType> = ({ todo, removeTask, completeTask }) => {

File: src/components/Todo/ToDo.tsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
+import React, { useState, useContext } from 'react'
+import { Context } from '../../context/context'
+import ToDoForm from '../ToDoForm'
+import ToDoList from '../ToDoList'
+
+const ToDo: React.FC = () => {
+  const initialTodo = useContext(Context)
```
Comment (SerhiienkoRuslan) [inline src/components/Todo/ToDo.tsx]: Do you need context here?

**Pull request merged**